src/main/java/dev/mcnees/moneymapper/batch/TransactionCleansingProcessor.java

```diff
@@ -0,0 +1,17 @@
+package dev.mcnees.moneymapper.batch;
+
+import dev.mcnees.moneymapper.domain.Transaction;
+
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TransactionCleansingProcessor implements ItemProcessor<Transaction, Transaction> {
+
+	@Override
+	public Transaction process(Transaction item) throws Exception {
+		item.setDescription(item.getDescription().replaceAll(",",""));
+		item.setAmount(-item.getAmount());
```

Comment:
This is getting into the 'Business Logic', or more commonly referred to here as 'Bob's preference'.

Line 13 removes commas because that messes with my CSV format.  I want a simple copy/paste from the outputted CSV.  I open that CSV in Excel and copy/paste directly to a larger `.xlsx` spreadsheet.  Any extra commas mess with the field count and the copy/paste, so I remove them.  I know there are other ways I could handle this with less common delimiters, but asking Microsoft to recognize those delimiters isn't great.  If I remember right delimiter choice is global to the machine, so if I switch Excel and `money-mapper` to use a pipe delimiter, I have to manually switch it back if I open a comma delimited CSV from somewhere else.

Line 14 is switching how the data is outputted.  It is my preference to see this as a spend:
Starbucks:  $3.49
Instead of this:
Starbucks:  -$3.49
Since almost every transaction is an outlay of cash, I remove the minus sign so it's easier for me to read.  Makes the cash flow calculations a little easier later on, IMO.

If this breaks the idempotent nature though then I need to give it a rethink.  My datasets are so small that the runs are very quick.  That and the output file is a cattle, not a pet.  I just overwrite the output file everytime regardless. 
 For those reasons I didn't consider retry functionality because I won't use it.  But I obviously don't want to put in an anti-pattern either.